kernel-scriptlets/rpm-script

```diff
@@ -115,6 +140,67 @@ update_bootloader() {
     fi
 }
 
+check_space_in_boot() {
+    # see bug #259303
+    # this script runs when the kernel gets updated with YaST
+    # YaST calls rpm always with -U
+    # -U replaces all packages with the new one
+    # rpm removes the files from the old packages after the postinstall script ran
+    # this will double the required space below /boot
+    # remove the files from the old packages to make room for the new initrd
+    # rpm may complain about low disk space if /boot/vmlinux does not fit
+    #
+    # When sdbootutil is used, the kernel and initrd should be living
+    # in sysroot, and transferred into /boot later by the snapper
+    # plugin (that calls sdbootutil add-all-kernels)
+    #
+    if [ "$YAST_IS_RUNNING" != "" ]; then
+	mydf="$( POSIXLY_CORRECT=1 df -P /boot/ | awk '/^(\/|-[[:blank:]])/{ print $4}' )"
+	if test "$mydf" != "" ; then
+	    echo "Free diskspace below /boot: $mydf blocks"
+	    # echo "512 byte blocks: $(( 2 * 1024 * 20 ))"
+	    if test "$mydf" -lt  "40960" ; then
+		echo "make room for new kernel '"$flavor"' because there are less than 20MB available."
+		# disabled because it breaks patch rpms
+		#rm -fv /boot/"$image"-*-"$flavor"
+		rm -fv /boot/initrd-*-"$flavor"
+	    fi
+	fi
+    fi
+}
+
+copy_or_link_legacy_files() {
+    # compat stuff for /boot.
+    # if /boot and /usr are not separate partitions we can just link
+    # the kernel there to save space. Otherwise copy.
+    if mountpoint -q /boot || mountpoint -q /usr; then
+	copy_or_link="cp -a --remove-destination"
+	separate_boot='1'
+    else
+	copy_or_link="ln -sf"
+	separate_boot=""
+    fi
+
+    for x in "$image" sysctl.conf System.map config; do
+	if [ "$separate_boot" = 1 ] || [ ! -e /boot/$x-"$kernelrelease"-"$flavor" ]; then
+	    $copy_or_link .."$modules_dir"/$x /boot/$x-"$kernelrelease"-"$flavor" || script_rc=$?
+	    if [ -e "$modules_dir"/.$x.hmac ]; then
+		$copy_or_link .."$modules_dir"/.$x.hmac /boot/.$x-"$kernelrelease"-"$flavor".hmac || script_rc=$?
+	    fi
+	fi
+    done
+
+    for x in /boot/"$image" /boot/initrd; do
+	rm -f "$x"
+	ln -s "${x##*/}-$kernelrelease-$flavor" "$x"
+    done
+    rm -f /boot/.vmlinuz.hmac
+    [ ! -e "/boot/.vmlinuz-$kernelrelease-$flavor.hmac" ] ||
+	ln -s ".vmlinuz-$kernelrelease-$flavor.hmac" /boot/.vmlinuz.hmac
+
+    return script_rc
```

Comment:
Ah ... I was confused by the fact that you're using the variable name `script_rc`, which is meant to be a global variable in the script.

We can't do that. If `script_rc` is used in this way, the variable should be renamed e.g. to `rc`, and `local` should be used.

**However**, I realize now that current code sets `script_rc` in `update_bootloader()`, in the same way in which you were using it in the original patch set. I guess this use is ok, then, for now at least. So please restore the original way of using `script_rc`.

Reply:
yes, you are right, `update_bootloader_entry()` uses a similar approach. I reverted back this part